bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/HSBType.java

```diff
@@ -71,18 +74,13 @@ public HSBType(DecimalType h, PercentType s, PercentType b) {
     }
 
     public HSBType(String value) {
-        if (value != null) {
```

Comment:
Isn't this inconsistent? For StringType we chose to allow `null` in the constructor and behave just like the default constructor was called. Here we do not allow `null` and throw an `IAE` instead.
Maybe logging a warning here is sufficient?
@maggu2810 any thoughts?

Reply:
One of the issues with still allowing null values is that it will hide errors in bindings so it makes using null annotations less effective.

Reply:
I can keep the null checks and throw IAEs for backwards compatibility. 

Though it will cause warnings about redundant null checks to be shown. These could be suppressed with `@SuppressWarnings({ "unused", "null" })` but that may hide real NPE issues well.

Reply:
@htreu As you have perhaps written in the linked issue, IMHO also `StringType` should not allow a nullable value :wink: So, IMHO we don't need to allow it here.

Reply:
I wanted to point out that we introduce inconsistent behaviour to the constructors of our `State` types: For `StringType` this PR introduces the `@Nullable` constructor which defaults to use the empty String internally.
For `HSBType` and `PointType` the value constructors are `@NonNull` annotated and will throw an exception if used with `null` values.

Reply:
Isn't it correct?
The `StringType` has been allowed to be called with `null` the `HSBType` and the `PointType` have not been allowed to be called with `null`. This behavior is not changed by @wborn just annotated correctly. And this is what this PR is about..

Reply:
@htreu We could discuss about a general handling for a constructor with a null reference on all our types but I would prefer to move that discussion into a separate PR. Personally I don't think it is good programming style. If you want to use a non explicit construction, don't use the constructor that takes e.g. a HSB value. If we want this behavior to allow lazy programming in rules that is (IMHO) again another story and it is more about script support.